webhooks/validating/configuration/src/main.rs

```diff
@@ -789,6 +860,256 @@ mod tests {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "valueFrom": {
+                }
+            }
+        ],"#;
+
+        // valueFrom discovery properties, not specified should fail, missing content of 'valueFrom'
+        run_validate_configuration_discovery_properties(discovery_properties);
+    }
+
+    #[test]
+    #[should_panic(expected = "Could not parse as Akri Configuration")]
+    fn test_validate_configuration_discovery_properties_unknown_value_from() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "valueFrom": {
+                    "fieldRef": {
+                        "fieldPath": "ffff"
+                    }
+                }
+            }
+        ],"#;
+
+        // valueFrom discovery properties, unknown ref should fail
+        run_validate_configuration_discovery_properties(discovery_properties);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_secret() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "valueFrom": {
+                    "secretKeyRef": {
+                        "name": "nnnn1",
+                        "key": "kkk",
+                        "optional": false
+                    }
+                }
+            }
+        ],"#;
+
+        // valueFrom discovery properties, secretKeyRef should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_config_map() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "valueFrom": {
+                    "configMapKeyRef": {
+                        "name": "nnnn1",
+                        "key": "kkk",
+                        "optional": false
+                    }
+                }
+            }
+        ],"#;
+
+        // valueFrom discovery properties, configMapKeyRef should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_value_and_value_from() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "value": "vvvv",
+                "valueFrom": {
+                    "configMapKeyRef": {
+                        "name": "nnnn1",
+                        "key": "kkk",
+                        "optional": false
+                    }
+                }
+            }
+        ],"#;
+
+        // both value and valueFrom are specified should success
```

Comment:
curious why setting value & valueFrom is valid?

Reply:
we don't restrict that, if both value and valueFrom are specified, value wins

Reply:
I had updated the schema so it's validated by by x-kubernetes-validations via OpenAPI Schema.  If both value and valueFrom are specified, the validator reject it.